terraform/slurm_cluster/modules/_slurm_instance/main.tf

```diff
@@ -153,3 +173,20 @@ resource "google_compute_instance_from_template" "slurm_instance" {
     replace_triggered_by = [null_resource.replace_trigger.id]
   }
 }
+
+##############
+# IP ADDRESS #
+##############
+
+resource "google_compute_address" "static_ip" {
+  for_each = {
+    for nic in flatten(
+      [for net_cfg in local.network_config : [for nic in net_cfg.network_interfaces : nic]]
+    ) : nic.ip_address_name => nic
+  }
+  name         = each.value.ip_address_name
+  subnetwork   = each.value.subnetwork
+  address_type = "INTERNAL"
+  region       = var.region
+  address      = each.value.network_ip
```

Comment:
`project` should be set. I guess it should be a project from subnetwork.
Caveat: we never provide `var.subnetwork_project`, instead `subnetwork` is selfLink, that is sufficient for the providers.
In your case you would need to:
* Rename `var.subnetwork -> var.subnetwork_self_link`
And either parse subnetwor selfLink or create data source to extract project_id.

Reply:
The project doesn't have to (or even can't?) [be in the SharedVPC](https://cloud.google.com/vpc/docs/reserve-static-internal-ip-address#sharedvpc). For sure what worked for me, creating them in service project. And I think it is better from management perspective (you see all the reserved IP addresses in service project).

But I'll do the renames and remove `subnetwork_project` and see how it works.

Reply:
Rename in `_slurm_instance` will require aligning the call sites of the module in cluster-toolkit.

Reply:
Per discussion, I rolled back renaming `subnetwork` to `subnetwork_self_link`, but I dropped the `subnetwork_project`.

Should I mark it as `DEPRECATED` instead of silently dropping, as it is done in Cluster Toolkit?